### PR TITLE
Cleanup Flyway Config

### PR DESCRIPTION
**Pull request by gastaldi**
- Migrate to `ConfigMapping`

### CODE REVIEW

File: extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java [hunk 1]
```diff
@@ -40,37 +28,32 @@ public static FlywayDataSourceRuntimeConfig defaultConfig() {
...
+    Optional<String> password();
 
     /**
      * Comma-separated case-sensitive list of schemas managed by Flyway.
      * The first schema in the list will be automatically set as the default one during the migration.
      * It will also be the one containing the schema history table.
      */
-    @ConfigItem
-    public Optional<List<String>> schemas = Optional.empty();
+    Optional<String[]> schemas();
```
Comment (geoand) [inline extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:56]: Why this change of types?

Comment (gastaldi) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:56]: Because `FluentConfiguration` expects a `String[]`.

See https://flywaydb.org/documentation/usage/api/javadoc/org/flywaydb/core/api/configuration/FluentConfiguration.html#schemas(java.lang.String...)

Comment (geoand) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java:56]: Sure, but we don't really use `String[]` so I find it odd

Comment (gastaldi) [reply extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java]: I reverted to `Optional<List<String>>` to avoid any unintentional type change

**Review by geoand — APPROVED** [review]
